Sources/HackernewsNLU/Controller.swift

```diff
@@ -189,75 +175,71 @@ public class Controller {
     ///
     /// - parameter completion: Completion closure invoked on success
     /// - parameter failure:    Failure closure invoked on error
-    func callNLUApi(newsArticle article: NewsArticles,completion: @escaping (JSON)->(), failure: @escaping (String) -> Void) -> Void {
-        let path = "/natural-language-understanding/api/v1/analyze?version=2017-02-27"
-        guard let username = nluCreds["username"] else {
-            failure("No username for NLU service")
-            return
-        }
-        guard let password = nluCreds["password"] else {
-            failure("No password for NLU service")
-            return
-        }
-        var requestOptions: [ClientRequest.Options] = []
-        var headers = [String:String]()
-        let theBody: JSON = [
-            "url": article.getLink(),
-            "features":[
-                "concepts":   [ "limit":10],
-                "categories":  [ "limit":10],
-                "emotion":  [ "limit":10],
-                "entities":  [ "limit":10],
-                "sentiment":  [ "limit":10],
-                "keywords":  [ "limit":10]
-            ]
-        ]
-        var theData = Data()
-        do {
-            try theData = theBody.rawData(options: JSONSerialization.WritingOptions())
+    func callNLUApi(newsArticle article: NewsArticles,completion: @escaping (SwiftyJSON.JSON)->(), failure: @escaping (String) -> Void) -> Void {
+        let service: NaturalLanguageUnderstanding!
+        if nluCreds["apikey"] == nil {
```

Comment:
`nluCreds` is set by calling `initService`. `initService` doesn't seem to handle the IAM creds. I think that `initService` needs to be updated for this if statement to ever go into the else block.